jdaviz/configs/imviz/tests/test_catalogs.py

```diff
@@ -208,110 +208,97 @@ def test_from_file_parsing(imviz_helper, tmp_path):
     assert catalogs_plugin.from_file_message == 'Table does not contain required sky_centroid column'  # noqa
 
 
-def test_offline_ecsv_catalog(imviz_helper, image_2d_wcs, tmp_path):
+def test_offline_ecsv_catalog(imviz_helper, image_2d_wcs):
     sky = SkyCoord(ra=[337.5202807, 337.51909197, 337.51760596],
                    dec=[-20.83305528, -20.83222194, -20.83083304], unit='deg')
     tbl = QTable({'sky_centroid': sky})  # Table has no "Label" column
-    tbl_file = str(tmp_path / 'sky_centroid.ecsv')
-    tbl.write(tbl_file, overwrite=True)
     n_entries = len(tbl)
 
     ndd = NDData(np.ones((10, 10)), wcs=image_2d_wcs)
     imviz_helper.load_data(ndd, data_label='data_with_wcs')
     assert len(imviz_helper.app.data_collection) == 1
 
-    catalogs_plugin = imviz_helper.plugins['Catalog Search']._obj
-    catalogs_plugin.from_file = tbl_file
-    catalogs_plugin.catalog_selected = 'From File...'
-    out_tbl = catalogs_plugin.search(error_on_fail=True)
+    catalogs_plugin = imviz_helper.plugins['Catalog Search']
+    catalogs_plugin.import_catalog(tbl)
+    out_tbl = catalogs_plugin._obj.search(error_on_fail=True)
     assert len(out_tbl) == n_entries
-    assert catalogs_plugin.number_of_results == n_entries
+    assert catalogs_plugin._obj.number_of_results == n_entries
```

Comment:
why the ._obj added here (and below)?

Reply:
Because I changed `catalogs_plugin = imviz_helper.plugins['Catalog Search']._obj` to `catalogs_plugin = imviz_helper.plugins['Catalog Search']` (old: L223, new: L221) so I can accessed the public API publicly, which means I need to throw in a `._obj` for all others. Hope this clarifies the matter.

Reply:
oh i missed that, sorry! thanks for clarifying